pkgs/development/compilers/llvm/15/default.nix

```diff
@@ -208,14 +208,17 @@ in let
       bintools = tools.bintools-unwrapped;
     };
 
-    clangUseLLVM = wrapCCWith rec {
+    clangUseLLVM =
+    if stdenv.targetPlatform.libc == null then
+      builtins.trace "yo" tools.clangNoLibc # TODO: gate correctly...
```

Comment:
@rrbutani what did you mean here for the gating? It seems like it's correct to me?

Reply:
I _think_ I wasn't sure whether we wanted to check for `libc == null` here or specifically exempt UEFI targets like we do in `all-packages.nix` (which will complain if it sees `libc == null` for a target it doesn't have special handling for):
https://github.com/NixOS/nixpkgs/blob/132b7981ef4ea88d9810ff286c1bee06bb8bf022/pkgs/top-level/all-packages.nix#L20381-L20404

---

But I think you're right and that this is fine.